src/metaopt/io/database/mongodb.py

```diff
@@ -19,95 +20,121 @@
+                                                 self._dbhost, self._dbport)
+            self._uri = uri
+        except:
+            raise ValueError("Incorrect values inserted for URI")
+
     def initiate_connection(self):
         try:
             conn = MongoClient(self.uri)
             self.conn = conn
         except pymongo.errors.ConnectionFailure:
             raise Exception("Could not connect to server")
+            self.logger.info("Could not connect to server")
         # return connection string for checking purposes
         return conn
 
     def check_connection(self):
         if self.db:
-            #print("Connection Successful!")
+            self.logger.info("Connection Successful!")
             return True
         else:
-            #print("Please check your connection")
+            self.logger.info("Please check your connection")
             return False
 
     def close_connection(self):
         try:
             close = self.conn.close()
         except pymongo.errors.ConnectionFailure:
             raise Exception("Could not close socket connection")
+            self.logger.info("Could not close socket connection")
         return close
 
     def is_locked(self):
         try:
             is_locked = self.conn.is_locked()
         except pymongo.errors.ConnectionFailure:
             raise Exception("Could not determine if connection is locked")
+            self.logger.info("Could not determine if connection is locked")
         return is_locked
 
     def unlock(self):
         try:
             unlock = self.conn.unlock()
         except pymongo.errors.ConnectionFailure:
-            raise Exception("Could determine unlock due to connection")
+            raise Exception("Could not determine unlock due to connection")
+            self.logger.info("Could not determine unlock due to connection")
+        return unlock
 
     def insert(self, experiment_name, trial_field, dictionary, field_filters=None):
 
-        # check field_filters value
         if field_filters is None:
             field_filters = {}
 
-        dbcollection    = self.db[experiment_name][trial_field]
+        dbcollection = self.db[experiment_name][trial_field]
         result = dbcollection.insert(field_filters, dictionary)
 
         return result
 
     def update(self, experiment_name, trial_field, update_dictionary, field_filters=None):
         # note here that update needs to have
         # MongoDB comparison operations
-        # for example update={"$set": {'hyperparam_name': 'hidden units',
-        #                               'hyperparam_type': 'int',
-        #                               'hyperparam_value': '200'}
+        # for example update=
+        #   {
+        #       "$set": {'hyperparam_name': 'hidden units',
+        #               'hyperparam_type': 'int',
+        #               'hyperparam_value': '200'
+        #               }
+        #   }
```

Comment:
That would be worth turning into a doctstring

Reply:
Cool